drivers/ethernet/nxp_imx_netc/eth_nxp_imx_netc_psi.c

```diff
@@ -216,8 +216,8 @@ static const struct ethernet_api netc_eth_api = {.iface_api.init = netc_eth_ifac
 		IF_DISABLED(CONFIG_ETH_NXP_IMX_NETC_MSI_GIC,                                       \
 			(.tx_intr_msg_data = NETC_TX_INTR_MSG_DATA_START + n,                      \
 			.rx_intr_msg_data = NETC_RX_INTR_MSG_DATA_START + n,))                     \
-		IF_ENABLED(CONFIG_PTP_CLOCK_NXP_NETC,				                   \
-			(.ptp_clock = DEVICE_DT_GET_OR_NULL(DT_INST_PHANDLE(n, ptp_clock)),))      \
+		IF_ENABLED(CONFIG_NET_L2_PTP, (IF_ENABLED(CONFIG_PTP_CLOCK_NXP_NETC,               \
```

Comment:
```suggestion
		IF_ENABLED(NETC_PTP_TIMESTAMPING_SUPPORT,               \
```
why not like this, like the others?

Reply:
Thanks. This is perfect.

Updated to v3. Changes include,
- Code improvement per @maass-hamburg suggestion.

Thanks.